rust/avdschema/src/schema/base/documentation_options.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2025 Arista Networks, Inc.
+// Use of this source code is governed by the Apache License 2.0
+// that can be found in the LICENSE file.
+
+use serde::{Deserialize, Serialize};
+use serde_with::skip_serializing_none;
+
+/// Special options used for generating documentation
+#[skip_serializing_none]
+#[derive(Debug, Default, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct DocumentationOptions {
+    /// Setting `table` will allow for custom grouping of schema fields in the documentation.
+    /// By default each root key has it's own table. By setting the same table-value on multiple keys, they will be merged to a single table.
+    /// If `table` is set on a 'child' key, all 'ancestor' keys are automatically included in the table so the full path is visible.
+    /// The `table` option is inherited to all child keys, unless specifically set on the child.
+    pub table: Option<String>,
+}
+
+/// Special options used for generating documentation
+#[skip_serializing_none]
+#[derive(Debug, Default, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct DocumentationOptionsDict {
```

Comment:
why do we need both structures (Dict and not Dict)

Reply:
hide_keys is in the one for dicts. We have the same in our metaschema today.